### PR TITLE
Adding comments for Auth classes and updating TOC

### PR DESCRIPTION
**Pull request by egilmorez**
After mistakenly creating a pull request into master, I hope to properly push my doc changes into hkj-typedoc :)

Thanks!

### CODE REVIEW

File: src/index.d.ts [hunk 1]
```diff
@@ -633,6 +1310,9 @@ declare namespace admin.auth {
     ): Promise<admin.auth.AuthProviderConfig>;
   }
 
+  /**
```
Comment (hiranya911) [inline src/index.d.ts]: Perhaps this empty block can be removed?

Comment (egilmorez) [reply src/index.d.ts]: You bet Hiranya -- staged at https://firebase.devsite.corp.google.com/docs/reference/admin/node/admin.auth.UserRecord.

As you noticed, I should clarify that the Auth methods commenting will come in a separate following pull request.  Thanks!

File: src/index.d.ts [hunk 2]
```diff
@@ -393,197 +415,852 @@ declare namespace admin.app {
...
+     */    
     emailVerified: boolean;
+
+    /**
+     * The user's display name.
+     */    
     displayName?: string;
+
+    /**
+     * The user's primary phone number, if set..
```
Comment (bojeil-google) [inline src/index.d.ts]: extra dot.

File: src/index.d.ts [hunk 3]
```diff
@@ -393,197 +415,852 @@ declare namespace admin.app {
...
     toJSON(): Object;
   }
 
+  /**
+   * Interface representing the properties to update on the provided user.
+   */ 
   interface UpdateRequest {
+
+    /**
+     * Whether or not Whether or not the user is disabled: `true` for disabled;
```
Comment (bojeil-google) [inline src/index.d.ts]: "Whether or not" is duplicated.

File: src/index.d.ts [hunk 4]
```diff
@@ -393,197 +415,852 @@ declare namespace admin.app {
...
+     * identifier for your Firebase project, which can be found in [your project's
+     * settings](https://console.firebase.google.com/project/_/settings/general/android:com.random.android).
+     */ 
     aud: string;
+
+    /**
+     * Time, in seconds since the Unix epoch, when the end-user authentication
+     * occurred.
+     *
+     * This value is not when this particular ID token was created, but when the
```
Comment (bojeil-google) [inline src/index.d.ts]: Seems like a verb is missing?
This valid is not "set" when...

File: src/index.d.ts [hunk 5]
```diff
@@ -393,197 +415,852 @@ declare namespace admin.app {
...
+       * Provider-specific identity details corresponding
+       * to the provider used to sign in the user.
+       */ 
       identities: {
         [key: string]: any;
       };
+
+      /**
+       * The ID of the provider used to sign in the
+       *   user. One of `"anonymous"`, `"password"`, `"facebook.com"`, `"github.com"`,
```
Comment (bojeil-google) [inline src/index.d.ts]: This block does not seem to align with the above statement. You have 2 additional spaces.

File: src/index.d.ts [hunk 6]
```diff
@@ -393,197 +415,852 @@ declare namespace admin.app {
...
+       *   `"google.com"`, `"twitter.com"`, or `"custom"`.
+       */ 
       sign_in_provider: string;
-      [key: string]: any;
+        [key: string]: any;
```
Comment (bojeil-google) [inline src/index.d.ts]: Indentation is unnecessary.

File: src/index.d.ts [hunk 7]
```diff
@@ -393,197 +415,852 @@ declare namespace admin.app {
...
+
+    /**
+     * The `uid` corresponding to the user who the ID token belonged to.
+     *
+     * This value is not actually in the JWT token claims itself. It is added as a
+     * convenience, and is set as the value of the [`sub`](#sub) property.
+     */
     uid: string;
-    [key: string]: any;
+     [key: string]: any;
```
Comment (bojeil-google) [inline src/index.d.ts]: Remove extra indentation

File: src/index.d.ts [hunk 8]
```diff
@@ -393,197 +415,852 @@ declare namespace admin.app {
...
+    /**
+     * An array of providers (for example, Google, Facebook) linked to the user.
+     */ 
     providerData?: {
       uid: string,
       displayName?: string,
       email?: string,
       photoURL?: string,
       providerId: string,
     }[];
```
Comment (bojeil-google) [inline src/index.d.ts]: I think we can replace with:
`providerData?: admin.auth.UserInfo[];`

File: src/index.d.ts [hunk 9]
```diff
@@ -393,197 +415,852 @@ declare namespace admin.app {
...
+     * The next page token, if available.
+     */
     pageToken?: string;
   }
 
+  
   type UpdateAuthProviderRequest =
       admin.auth.SAMLUpdateAuthProviderRequest | admin.auth.OIDCUpdateAuthProviderRequest;
-
+
```
Comment (bojeil-google) [inline src/index.d.ts]: Remove extra spaces

File: src/index.d.ts (hunk 5 above)
Comment (bojeil-google) [reply src/index.d.ts]: this is still not aligned.

**Pull request merged**